MEC15xx/MEC1501: deprecate pinmux and move to pinctrl

jvasanth1 opened the pull request:
Update MEC15xx/MEC1501 drivers to use pinctrl instead of pimux. Adding drivers one by one starting with ps2 and kscan.

jvasanth1 committed:
ps2: mec15xx: add pinctrl for mec15xx/mec1501 ps2

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 ps2. Update board dts and pinmux files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 committed:
kscan: mec15xx: add pinctrl for mec15xx/mec1501 kscan

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 kscan. Update board dts and pinmux files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

gmarull commented on boards/arm/mec1501modular_assy6885/mec1501modular_assy6885.dts:
```diff
@@ -81,10 +81,30 @@
+	pinctrl-names = "default";
+};
+
+&ps2_clk0b_gpio007 {
+	drive-open-drain;
+};
```
as suggested in the pinctrl guide, customization of pre-defined nodes should happen in a `board-pinctrl.dtsi` file.

jvasanth1 replied:
added `board-pinctrl.dtsi` file

jvasanth1 committed:
pwm: mec15xx: add pinctrl for mec15xx/mec1501 pwm

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 pwm. Update board dts and pinmux files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

gmarull commented:
ping

gmarull commented:
@jvasanth1  +1 ping

henrikbrixandersen commented:
Any progress on this? The deadline for v3.2 is approaching fast.

gmarull commented:
> Any progress on this? The deadline for v3.2 is approaching fast.

@jvasanth1 is out until 12th, so we either remove the boards for now or ask for an exception…

fabiobaltieri commented:
> > Any progress on this? The deadline for v3.2 is approaching fast.
> 
> @jvasanth1 is out until 12th, so we either remove the boards for now or ask for an exception…

This is for unblocking the pinmux common code cleanup right? Maybe we can leave the board and just remove the pinmux driver, and mark the corresponding board as known non functional?

gmarull commented:
> > > Any progress on this? The deadline for v3.2 is approaching fast.
> > 
> > 
> > @jvasanth1 is out until 12th, so we either remove the boards for now or ask for an exception…
> 
> This is for unblocking the pinmux common code cleanup right? Maybe we can leave the board and just remove the pinmux driver, and mark the corresponding board as known non functional?

I'd advocate for temporary removal, having a broken board upstream doesn't sound like a good idea.

fabiobaltieri commented:
> > > > Any progress on this? The deadline for v3.2 is approaching fast.
> > > 
> > > 
> > > @jvasanth1 is out until 12th, so we either remove the boards for now or ask for an exception…
> > 
> > 
> > This is for unblocking the pinmux common code cleanup right? Maybe we can leave the board and just remove the pinmux driver, and mark the corresponding board as known non functional?
> 
> I'd advocate for temporary removal, having a broken board upstream doesn't sound like a good idea.

Cool, it's a very strong signal, should we flag this for TSC discussion for now?

gmarull commented:
> > > > > Any progress on this? The deadline for v3.2 is approaching fast.
> > > > 
> > > > 
> > > > @jvasanth1 is out until 12th, so we either remove the boards for now or ask for an exception…
> > > 
> > > 
> > > This is for unblocking the pinmux common code cleanup right? Maybe we can leave the board and just remove the pinmux driver, and mark the corresponding board as known non functional?
> > 
> > 
> > I'd advocate for temporary removal, having a broken board upstream doesn't sound like a good idea.
> 
> Cool, it's a very strong signal, should we flag this for TSC discussion for now?

Ok

jvasanth1 commented:
> @jvasanth1 would you be able to prioritize finishing this PR? It's blocking pinmux deprecation and there are conversations about disabilng the boards from CI or remving them entirely. Would you be able to complete this before RC2? (that's next week)

will take this as priority once I am back early next week and try to close by end of next week

mmahadevan108 commented:
@jvasanth1 any update on this PR?

jvasanth1 committed:
peci: mec15xx: add pinctrl for mec15xx/mec1501 peci

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 peci. Update board dts and pinmux files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 committed:
tach: mec15xx: add pinctrl for mec15xx/mec1501 tach

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 tach. Update board dts and pinmux files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 commented:
> @jvasanth1 any update on this PR?

working on this PR and adding more commits

jvasanth1 committed:
soc: arm: microchip: update mec1501 soc header

update mec1501 soc.h to include soc_dt.h for pinctrl
related macros

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 committed:
espi: mec15xx: add pinctrl for mec15xx/mec1501 espi

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 espi. Update board dts, pinmux and espi driver
files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

mmahadevan108 commented:
@jvasanth1 we are close to RC2 release date, when do you think this will be done?

jvasanth1 commented:
> @jvasanth1 we are close to RC2 release date, when do you think this will be done?

I am working on it but looks like I need more time to complete this effort.

jvasanth1 committed:
i2c: mec15xx: add pinctrl for mec15xx/mec1501 i2c

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 pwm. Update board dts, pinmux and driver files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

mmahadevan108 commented:
@jvasanth1, can you help fix the CI failures on this PR. We can then merge this in and you can add an issue to complete any pending changes.

jvasanth1 committed:
mec15xx: enable pinctrl for mec15xx/mec1501 boards

enable CONFIG_PINCTRL for mec15xx and mec1501 boards

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

str4t0m commented on boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig:
```diff
@@ -17,5 +17,6 @@ CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_PINMUX=y
```
could you please remove `CONFIG_PINMUX=y`

jvasanth1 replied:
pinmux is not yet completely removed, can be removed only at last once all drivers stop using pinmux, this PR is still work in progress

str4t0m replied:
I see, sorry my bad.

str4t0m commented on boards/arm/mec15xxevb_assy6853/mec15xxevb_assy6853_defconfig:
```diff
@@ -15,6 +15,7 @@ CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_PINMUX=y
```
same here

str4t0m reviewed (commented):
Hi, it seems the CI failures are all gone now.
I just skimmed over board files and noticed the CONFIG_PINMUX are still left.

str4t0m commented on boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig:
```diff
@@ -17,5 +17,6 @@ CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_PINMUX=y
 CONFIG_GPIO=y
```
and here.

mmahadevan108 commented:
@jvasanth1, can you let us know what items are remaining?
@gmarull, can you take a look at this PR.

jvasanth1 commented:
> @jvasanth1, can you let us know what items are remaining?

adc and qmspi drivers are remaining and mec172x uart driver needs update to work with mec15xx as well. Also I need to review all other pins configured currently in pinmux.c

mmahadevan108 commented:
@jvasanth1, thank you for your updates. Among the pending work, can we prioritize updating mec172x uart driver.

fabiobaltieri commented:
@jvasanth1 in the interest of moving this forward, how about adding one more patch dropping `pinctrl.c` and `CONFIG_PINMUX=y` (you can cherry-pick https://github.com/zephyrproject-rtos/zephyr/pull/50442) and the submitting the whole thing? That would allow us to then proceed with deprecating the pinmux API, and would leave these two board semi-functional with the peripheral ported already, then on a second stage you can send a fixup PR for the remaining peripheral, of we can flag those as not working under "known issues" for 3.2.0 and then fix them in 3.2.1 once the patch is posted.

jvasanth1 commented:
> @jvasanth1 in the interest of moving this forward, how about adding one more patch dropping `pinctrl.c` and `CONFIG_PINMUX=y` (you can cherry-pick #50442) and the submitting the whole thing? That would allow us to then proceed with deprecating the pinmux API, and would leave these two board semi-functional with the peripheral ported already, then on a second stage you can send a fixup PR for the remaining peripheral, of we can flag those as not working under "known issues" for 3.2.0 and then fix them in 3.2.1 once the patch is posted.

ok, I will cherry pick #50422 commit and try to add adc and qmspi drivers as well today

jvasanth1 committed:
adc: mec15xx: add pinctrl for mec15xx/mec1501 adc

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 adc. Update board dts, pinmux and driver files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

fabiobaltieri commented:
> > @jvasanth1 in the interest of moving this forward, how about adding one more patch dropping `pinctrl.c` and `CONFIG_PINMUX=y` (you can cherry-pick #50442) and the submitting the whole thing? That would allow us to then proceed with deprecating the pinmux API, and would leave these two board semi-functional with the peripheral ported already, then on a second stage you can send a fixup PR for the remaining peripheral, of we can flag those as not working under "known issues" for 3.2.0 and then fix them in 3.2.1 once the patch is posted.
> 
> ok, I will cherry pick #50422 commit and try to add adc and qmspi drivers as well today

Cool, it's #50442 btw.

mmahadevan108 commented:
@jvasanth1. Instead of you cherry picking https://github.com/zephyrproject-rtos/zephyr/pull/50442 we would prefer the `pinmux.c` file delete is done as part of this PR. Can you help make this change, we would like to get in whatever is functional at the earliest along with delete of `pinmux.c`

You can then create a separate PR to work on the remaining features.

jvasanth1 commented:
> @jvasanth1. Instead of you cherry picking #50442 we would prefer the `pinmux.c` file delete is done as part of this PR. Can you help make this change, we would like to get in whatever is functional at the earliest along with delete of `pinmux.c`
> 
> You can then create a separate PR to work on the remaining features.

I am unable the cherry-pick anyway as my local workspace doesn't seem see the external commit. 
I will delete the `pinmux.c` as part of this PR.

fabiobaltieri commented:
> > @jvasanth1. Instead of you cherry picking #50442 we would prefer the `pinmux.c` file delete is done as part of this PR. Can you help make this change, we would like to get in whatever is functional at the earliest along with delete of `pinmux.c`
> > You can then create a separate PR to work on the remaining features.
> 
> I am unable the cherry-pick anyway as my local workspace doesn't seem to see the external commit. I will delete the `pinmux.c` as part of this PR.

Yeah that works, thanks (and the two config lines as well)

mmahadevan108 commented:
> > @jvasanth1. Instead of you cherry picking #50442 we would prefer the `pinmux.c` file delete is done as part of this PR. Can you help make this change, we would like to get in whatever is functional at the earliest along with delete of `pinmux.c`
> > You can then create a separate PR to work on the remaining features.
> 
> I am unable the cherry-pick anyway as my local workspace doesn't seem to see the external commit. I will delete the `pinmux.c` as part of this PR.

Great, please give this a priority. Thank you.

jvasanth1 committed:
spi: mec15xx: add pinctrl for mec15xx/mec1501 qmspi

Remove pinmux calls and add pinctrl support for mec15xx
and mec1501 qmspi. Update board dts, pinmux and driver files.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 committed:
boards: microchip: mec15xx: delete pinmux.c files

remove pinmux support in mec15xxevb and mec1501modular
boards. pinmux API is deprecated in 3.2 release.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 commented:
added commits for adc and spi driver pinctrl and removed pinmux.c

mmahadevan108 commented:
@jvasanth1 Please take a look at this CI failure:
```
198/625 mec1501modular_assy6885   samples/drivers/espi/sample.drivers.espi           FAILED Cmake build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/mec1501modular_assy6885/samples/drivers/espi/sample.drivers.espi/build.log
ERROR   - Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/samples/drivers/espi
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /__w/zephyr/zephyr/.cache
-- Zephyr version: 3.2.0-rc2 (/__w/zephyr/zephyr)
-- Found west (found suitable version "0.13.1", minimum required is "0.7.1")
-- Board: mec1501modular_assy6885
-- Found host-tools: zephyr 0.15.0 (/opt/toolchains/zephyr-sdk-0.15.0)
-- Found toolchain: zephyr 0.15.0 (/opt/toolchains/zephyr-sdk-0.15.0)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.15.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/mec1501modular_assy6885/mec1501modular_assy6885.dts
-- Found devicetree overlay: /__w/zephyr/zephyr/samples/drivers/espi/boards/mec1501modular_assy6885.overlay
devicetree error: 'pinctrl-0' is marked as required in 'properties:' in /__w/zephyr/zephyr/dts/bindings/spi/microchip,xec-qmspi.yaml, but does not appear in <Node /soc/spi@40070000 in '/__w/zephyr/zephyr/misc/empty_file.c'>
```

jvasanth1 committed:
samples: espi: add pinctrl in mec1501 board overlay

update mec1501modular board overlay to include pinctrl
properties for spi

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

mmahadevan108 commented:
@jvasanth1, thank you for your prompt attention to this PR. Can you please let us know what portions were not addressed as part of PR?

mmahadevan108 commented:
> > @jvasanth1, thank you for your prompt attention to this PR. Can you please let us know what portions were not addressed as part of PR?
> 
> There are mainly 2 things remaining:
> 
> 1. mec15xx didn't have a uart driver, the uart pins were configured in pinmux.c and uart worked with default ns16550 driver. Since there is no pinmux now to configure the uart pins, it needs to be done in a uart driver. So mec172xx uart driver needs to be updated to work for mec15xx as well.

You should'nt need a new uart driver. You can link the `uart pinctrl` pins into the uart node in the board `dts` file

mmahadevan108 commented:
Of course the `ns16550` UART driver will need to be updated to use `pinctrl`
